[mdnsresponder] Fix build with dynamic CRT

davidebeatrici opened the pull request:
mdnsresponder was being built with static CRT even when `VCPKG_CRT_LINKAGE` was set to `dynamic`:

```
-- Performing post-build validation
Expected Debug,Dynamic crt linkage, but the following libs had invalid crt linkage:

    C:/vcpkg/packages/mdnsresponder_x64-windows-static-md/debug/lib/dnssd.lib: Debug,Static

To inspect the lib files, use:
    dumpbin.exe /directives mylibfile.lib
Expected Release,Dynamic crt linkage, but the following libs had invalid crt linkage:

    C:/vcpkg/packages/mdnsresponder_x64-windows-static-md/lib/dnssd.lib: Release,Static

To inspect the lib files, use:
    dumpbin.exe /directives mylibfile.lib
Found 2 error(s). Please correct the portfile:
    C:\vcpkg\ports\mdnsresponder\portfile.cmake
-- Performing post-build validation done
Error: Building package mdnsresponder:x64-windows-static-md failed with: POST_BUILD_CHECKS_FAILED
```

msftclas commented:
[![CLA assistant check](https://cla.opensource.microsoft.com/pull/badge/not_signed)](https://cla.opensource.microsoft.com/microsoft/vcpkg?pullRequest=8953) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/vcpkg?pullRequest=8953) before we can accept your contribution.<br/><br/>:x: davidebeatrici [sign now](https://cla.opensource.microsoft.com/microsoft/vcpkg?pullRequest=8953)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.microsoft.com/check/microsoft/vcpkg?pullRequest=8953) it.</sub>

NancyLi1013 commented:
/azp run

NancyLi1013 commented:
Hi @davidebeatrici thanks for your PR.
Could you please also update the version in `CONTROL `file?

davidebeatrici committed:
mdnsresponder: fix build with dynamic CRT

mdnsresponder was being built with static CRT even when VCPKG_CRT_LINKAGE was set to "dynamic":

-- Performing post-build validation
Expected Debug,Dynamic crt linkage, but the following libs had invalid crt linkage:

    C:/vcpkg/packages/mdnsresponder_x64-windows-static-md/debug/lib/dnssd.lib: Debug,Static

To inspect the lib files, use:
    dumpbin.exe /directives mylibfile.lib
Expected Release,Dynamic crt linkage, but the following libs had invalid crt linkage:

    C:/vcpkg/packages/mdnsresponder_x64-windows-static-md/lib/dnssd.lib: Release,Static

To inspect the lib files, use:
    dumpbin.exe /directives mylibfile.lib
Found 2 error(s). Please correct the portfile:
    C:\vcpkg\ports\mdnsresponder\portfile.cmake
-- Performing post-build validation done
Error: Building package mdnsresponder:x64-windows-static-md failed with: POST_BUILD_CHECKS_FAILED

davidebeatrici commented:
I also rebased against `master`.

NancyLi1013 commented:
/azp run

ras0219-msft commented:
LGTM, thanks for the PR!

davidebeatrici commented:
You're welcome!
